### ISSUE TITLE
False positive in trivially_copy_pass_by_ref where reference is used

### ISSUE THREAD

**Issue by ogham** [rust-lang/rust-clippy#5645]
I've encountered a false positive with the [trivially_copy_pass_by_ref](https://rust-lang.github.io/rust-clippy/master/index.html#trivially_copy_pass_by_ref) lint.

```rust
    pub fn contains(&self, target: &u8) -> bool {
        exists_within(self.bytes, target)
    }
}

/// Given a `slice`, and a `reference` to a value, determine if the reference
/// is pointing to one of the values in the slice. This is a safe operation
/// because we can compare pointers without dereferencing anything.
pub fn exists_within<'a, T>(slice: &'a [T], reference: &'a T) -> bool {
}
```

I made a quick [Rust playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=451f1147a3ac5e15d33fa7eebf3af98c) with the code that ran into this issue.

Thanks for Clippy!

---

```
$ cargo clippy -V
clippy 0.0.212 (204bb9b5 2020-03-17)
```

**Comment by flip1995** [comment]
https://github.com/rust-lang/rust-clippy/blob/7ca335a97dab0869ec6e407a15bea86a7c7c629e/clippy_lints/src/trivially_copy_pass_by_ref.rs#L118

This should be `MaybeIncorrect`

To fix this issue and suggest for every occurrence of the parameter inside a function to add a `&` if necessary, is really hard. Also I'm not really sure if this would be worth it, since it would convolute the suggestion really hard.
